### PR TITLE
ci: fix setuptools-scm deploy

### PR DESCRIPTION
**Pull request by redeboer**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -54,15 +54,14 @@ jobs:
...
-      install: echo "Skipping install stage"
+      install:
+        - pip3 install .
       script: echo "Skipping script stage"
       deploy:
         edge: true
         provider: pypi
         on:
           tags: true
-        distributions: sdist
```
Comment (redeboer) [inline .travis.yml:64]: Removed these because afaics we should the travis defaults should be fine for `setuptools-scm`

File: .travis.yml [hunk 2]
```diff
@@ -54,15 +54,14 @@ jobs:
       language: shell
 
     - name: "Deploy PyPI package"
-      install: echo "Skipping install stage"
+      install:
```
Comment (redeboer) [inline .travis.yml:57]: I don't think installing is necessary, but at least I'm sure this creates a version file. Not sure what the `deploy` stage with `provider: pypi` does exactly.

File: .travis.yml [hunk 3]
```diff
@@ -54,15 +54,14 @@ jobs:
...
+        - pip3 install .
       script: echo "Skipping script stage"
       deploy:
         edge: true
         provider: pypi
         on:
           tags: true
-        distributions: sdist
-        skip_existing: false
```
Comment (spflueger) [inline .travis.yml:65]: Why are those removed?

File: .travis.yml (hunk 2 above)
Comment (spflueger) [reply .travis.yml:57]: Where does it create the version file in the install location?

File: .travis.yml (hunk 1 above)
Comment (spflueger) [reply .travis.yml:64]: Hmm i see. Maybe its better to keep the `skip_existing` in case they change the default value of that

File: .travis.yml (hunk 2 above)
Comment (redeboer) [reply .travis.yml:57]: https://github.com/ComPWA/expertsystem/blob/4fc2df6c21dd0c9fe611144b1de8d01b77a238c0/pyproject.toml#L9-L10

Comment (spflueger) [reply .travis.yml:57]: can't you test that the version will be correct locally. In addition you can also use https://test.pypi.org/ to upload

Comment (redeboer) [reply .travis.yml:57]: Thing is that I'm not sure what this `deploy` job does exactly on Travis.

Comment (spflueger) [reply .travis.yml:57]: well build the sdist and upload it to pypi

Comment (redeboer) [reply .travis.yml:57]: > This feature requires Setuptools 42 or later, released in Nov, 2019. **If your project needs to support build from sdist on older versions of Setuptools, you will need to also implement the `setup.py usage` for those legacy environments.**

Not sure what Travis does it you remove `sdist`

Comment (redeboer) [reply .travis.yml:57]: > well build the sdist and upload it to pypi

but yeah `sdist` does not work, also locally, it does not produce the version file

Comment (spflueger) [reply .travis.yml:57]: So does the current code work? I want to avoid doing trail and error here

Comment (redeboer) [reply .travis.yml:57]: No, also not `bdist_wheel`

Comment (redeboer) [reply .travis.yml:57]: Should work now :crossed_fingers:
